vpd-tool/include/tool_utils.hpp

```diff
@@ -125,5 +125,63 @@ inline std::string getPrintableValue(const types::BinaryVector& i_keywordValue)
 
     return l_oss.str();
 }
+
+/**
+ * @brief API to read keyword's value from hardware.
+ *
+ * This API reads keyword's value by requesting DBus service(vpd-manager) who
+ * hosts the 'ReadKeyword' interface to read keyword's value.
+ *
+ * @param[in] i_serviceName - Name of the Dbus service hosting 'ReadKeyword'
+ * method.
+ * @param[in] i_objectPath - Object path under the service.
+ * @param[in] i_interface - Interface under which 'ReadKeyword' method exist.
+ * @param[in] i_eepromPath - EEPROM file path.
+ * @param[in] i_paramsToReadData - Property whose value has to be read.
+ *
+ * @return - Value read from Dbus
+ *
+ * @return std::runtime_error
+ */
+inline types::DbusVariantType readKeywordFromHardware(
+    const std::string& i_serviceName, const std::string& i_objectPath,
+    const std::string& i_interface, const std::string& i_eepromPath,
+    const types::ReadVpdParams i_paramsToReadData)
+{
+    // Mandatory fields to make a dbus call.
+    if (i_serviceName.empty() || i_objectPath.empty() || i_interface.empty() ||
+        i_eepromPath.empty())
+    {
+        std::cout
+            << "One of the parameter to make DBus ReadKeyword call is empty."
+            << std::endl;
+        throw std::runtime_error("Empty Parameter");
+    }
+
+    try
+    {
+        types::DbusVariantType l_propertyValue;
+
+        auto l_bus = sdbusplus::bus::new_default();
+
+        auto l_method =
+            l_bus.new_method_call(i_serviceName.c_str(), i_objectPath.c_str(),
+                                  i_interface.c_str(), "ReadKeyword");
+
+        l_method.append(i_eepromPath, i_paramsToReadData);
+        auto l_result = l_bus.call(l_method);
+
+        l_result.read(l_propertyValue);
+
+        return l_propertyValue;
+    }
+    catch (const sdbusplus::exception::SdBusError& l_error)
+    {
+        // TODO: Enable logging only when verbose is enabled.
+        std::cout << l_error.what() << std::endl;
```

Comment:
Duplicate log message? We are logging here and also throwing an exception with the same message.

Reply:
yes, duplicate logs.
removed this log